lib/js/src/rpc/structs/VideoStreamingCapability.js

```diff
@@ -32,22 +32,34 @@
 
 import { RpcStruct } from '../RpcStruct.js';
 import { VideoStreamingFormat } from './VideoStreamingFormat.js';
+import { ImageResolution } from './ImageResolution.js';
 
 class VideoStreamingCapability extends RpcStruct {
+    /**
+    * @constructor
+    */
+    constructor () {
+        super();
+    }
+    
     /**
     * @param {ImageResolution} val
     * @return {VideoStreamingCapability}
     */
     setPreferredResolution (val) {
-        this.setParameter(VideoStreamingCapability.KEY_PREFERRED_RESOLUTION, val);
-        return this;
+        const errorMessage = 'val must be an instance of ImageResolution';
+        if (val instanceof ImageResolution) {
+            this.setParameter(VideoStreamingCapability.KEY_PREFERRED_RESOLUTION, val);
+            return this;
+        }
+        throw new Error(errorMessage);
```

Comment:
Validation should be done via the `validateType` method provided in the base `RpcStruct` class.